src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+                operatorKickParams[i].operatorIdsToSwap, 
+                operatorKickParams[i].globalOperatorListIndex
+            );
+        }
+    }
+
+    /**
+     * @notice Deregisters the msg.sender as an operator from the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registered for
+     * @param deregistrationData is the the data that is decoded to get the operator's deregisteration information
+     * @dev `deregistrationData` should be a tuple of the operator's BLS public key, the list of operator ids to swap, 
+     * and the operator's index in the global operator list
+     */
+    function deregisterOperatorWithCoordinator(bytes calldata quorumNumbers, bytes calldata deregistrationData) external {
+        // get the operator's deregisteration information
+        (BN254.G1Point memory pubkey, bytes32[] memory operatorIdsToSwap, uint32 globalOperatorListIndex) 
+            = abi.decode(deregistrationData, (BN254.G1Point, bytes32[], uint32));
+        // call internal function to deregister the operator
+        _deregisterOperatorWithCoordinator(msg.sender, quorumNumbers, pubkey, operatorIdsToSwap, globalOperatorListIndex);
+    }
+
+    /**
+     * @notice Deregisters the msg.sender as an operator from the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registered for
+     * @param pubkey is the BLS public key of the operator
+     * @param operatorIdsToSwap is the list of the operator ids that the should swap for the deregistering operator's index
+     * @param globalOperatorListIndex is the operator's index in the global operator list in the IndexRegistry
+     */
+    function deregisterOperatorWithCoordinator(bytes calldata quorumNumbers, BN254.G1Point memory pubkey, bytes32[] memory operatorIdsToSwap, uint32 globalOperatorListIndex) external {
+        _deregisterOperatorWithCoordinator(msg.sender, quorumNumbers, pubkey, operatorIdsToSwap, globalOperatorListIndex);
+    }
+
+    function _registerOperatorWithCoordinator(address operator, bytes calldata quorumNumbers, BN254.G1Point memory pubkey) internal {
+        require(
+            slasher.contractCanSlashOperatorUntilBlock(operator, address(serviceManager)) == type(uint32).max,
+            "StakeRegistry._registerOperator: operator must be opted into slashing by the serviceManager"
+        );
+        
+        // check that the sender is not already registered
+        require(_operators[operator].status != OperatorStatus.REGISTERED, "BLSIndexRegistryCoordinator._registerOperatorWithCoordinator: operator already registered");
+
+        // get the quorum bitmap from the quorum numbers
+        uint192 quorumBitmap = uint192(BytesArrayBitmaps.orderedBytesArrayToBitmap_Yul(quorumNumbers));
+        require(quorumBitmap != 0, "BLSIndexRegistryCoordinator._registerOperatorWithCoordinator: quorumBitmap cannot be 0");
```

Comment:
this check on the input is inadequate, given we pass on the `quorumNumbers`.
I think we should validate that the library output is `<= type(uint192).max`
This check should be repeated (maybe make a little function for this?) anywhere that the `quorumNumbers` are similarly supplied.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/2d4a033ef22fdb60df4b1bdb26907986e3f6a44a